pdns/ixfrdist.cc

```diff
@@ -194,8 +194,13 @@ void updateThread() {
     }
     time_t now = time(nullptr);
     for (const auto &domain : g_domains) {
-      if ((g_soas.find(domain) != g_soas.end() && now - lastCheck[domain] < g_soas[domain]->d_st.refresh) || // Only check if we have waited `refresh` seconds
-          (g_soas.find(domain) == g_soas.end() && now - lastCheck[domain] < 30))  {                          // Or if we could not get an update at all still, every 30 seconds
+      shared_ptr<SOARecordContent> current_soa;
+      if (g_soas.find(domain) != g_soas.end()) {
+        std::lock_guard<std::mutex> guard(g_soas_mutex);
```

Comment:
`g_soas.find()` is still accessing the `map` without holding the lock. In that case it might be fine because we are in the `updateThread` thread which is the only one updating the `map`, but it might be an issue in the future.